Sources/GRPCHTTP2TransportNIOPosix/NIOSSL+GRPC.swift

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#if canImport(NIOSSL)
+import NIOSSL
+
+extension NIOSSLSerializationFormats {
+  fileprivate init(_ format: HTTP2ServerTransport.Config.TLS.SerializationFormat) {
+    switch format.wrapped {
+    case .pem:
+      self = .pem
+    case .der:
+      self = .der
+    }
+  }
+}
+
+extension NIOSSLCertificateSource {
+  fileprivate static func certificateChain(
+    from certificateChainSources: [HTTP2ServerTransport.Config.TLS.CertificateSource]
+  ) throws -> [NIOSSLCertificateSource] {
```

Comment:
I find this API a little odd because it doesn't take `NIOSSLCertificateSource` as input or return `NIOSSLCertificateSource` rather an array of them.

Seems more appropriate as extension on `Sequence<HTTP2ServerTransport.Config.TLS.CertificateSource>`:

```swift
extension Sequence<HTTP2ServerTransport.Config.TLS.CertificateSource> {
  func sslCertificateSources() -> [NIOSSLCertificateSource] { 
    // ...
  }
}
```

Reply:
Would you say this is more idiomatic than a `fileprivate init` as an extension of `Array<NIOSSLCertificateSource>`?
I honestly don't have a strong preference here but only asking because we generally prefer initialisers for `Bar` instead of `Foo.toBar()`.